creator/ingest_runs/genomic_data_loader.py

```diff
@@ -0,0 +1,397 @@
+
+        return genomic_df
+
+    def load_specimen_harmonized_gf_links(self, genomic_df):
+        """
+        Step 2
+
+        Link the harmonized genomic files to their related specimens by loading
+        the appropriate biospecimen-genomic-file records into the Data Service
+        """
+        # Prepare data
+        biospec_gen_df = genomic_df[[
+            CONCEPT.GENOMIC_FILE.ID,
+            CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID,
+            CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID,
+        ]]
+        biospec_gen_df[CONCEPT.BIOSPECIMEN.ID] = biospec_gen_df[
+            CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID
+        ]
+        biospec_gen_df[CONCEPT.BIOSPECIMEN_GENOMIC_FILE.VISIBLE] = True
+        df = biospec_gen_df.drop_duplicates(
+            [CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID,
+             CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID],
+        )
+
+        # Load into Data Service
+        self.logger.info(
+            f"Creating {df.shape[0]} biospecimen-genomic-file links in "
+            f"{DATASERVICE_URL}"
+        )
+        _ = self._load_entities(BIO_GEN_FILE, df)
+
+        return biospec_gen_df
+
+    def load_seq_exp_harmonized_genomic_files(self, genomic_df):
+        """
+        Step 3
+
+        Link the harmonized genomic files to their related sequencing
+        experiments by loading the appropriate
+        sequencing-experiment-genomic-file records into the Data Service
+
+        This will have to be done by getting the sequencing experiments that
+        are linked to the source genomic files used to produce the harmonized
+        genomic files.
+        """
+        # Get sequencing experiments for the unharmonized genomic files in
+        # the study
+        se_source_gf_df = self._get_seq_experiment_genomic_files(
+            harmonized=False
+        )
+        # Merge ^ with the genomic df which as both harmonized and unharmonized
+        # gfs in one table
+        se_gf_df = pd.merge(
+            genomic_df, se_source_gf_df, on=CONCEPT.GENOMIC_FILE.SOURCE_FILE
+        )
+        se_gf_df[CONCEPT.SEQUENCING_GENOMIC_FILE.VISIBLE] = True
+
+        # Load the genomic-file sequencing-experiment links
+        df = se_gf_df.drop_duplicates(
+            [CONCEPT.SEQUENCING.TARGET_SERVICE_ID,
+             CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID],
+        )
+        self.logger.info(
+            f"Creating {df.shape[0]} sequencing-experiment-genomic-file "
+            f"links in {DATASERVICE_URL}"
+        )
+        _ = self._load_entities(SEQ_EXP_GEN_FILE, df)
+
+        return se_gf_df
+
+    def _prep_harmonized_gf_df(self, df):
+        """
+        Part of Step 1
+
+        Clean up the genomic file DataFrame (output from get_s3_file_metadata)
+        and get it ready to be ingested by transforming column values and
+        standardizing the column names.
+        """
+        self.logger.info("Preparing genomic file DataFrame for ingest ...")
+
+        def file_ext(x):
```

Comment:
Added this to extract the file extension/format. This is what the s3_scrape_config.py extract config uses. We should probably put this in a common module and then import it here rather than define it again